radio_beam/commonbeam.py

```diff
@@ -0,0 +1,522 @@
+
+import numpy as np
+import astropy.units as u
+
+try:
+    from scipy import optimize as opt
+    from scipy.spatial import ConvexHull
+    HAS_SCIPY = True
+except ImportError:
+    HAS_SCIPY = False
+
+from .beam import Beam
+from .utils import BeamError, transform_ellipse
+
+
+def commonbeam(beams, method='pts', **method_kwargs):
+    '''
+    Use analytic method if there are only two beams. Otherwise use constrained
+    optimization to find the common beam.
+    '''
+
+    if beams.size == 1:
+        return beams[0]
+    elif fits_in_largest(beams):
+        return beams.largest_beam()
+    else:
+        if beams.size == 2:
+            try:
+                return common_2beams(beams)
+            # Sometimes this method can fail. Use the many beam solution in
+            # this case
+            except (ValueError, BeamError):
+                pass
+
+        if method == 'pts':
+            return common_manybeams_mve(beams, **method_kwargs)
+        elif method == 'opt':
+            return common_manybeams_opt(beams, **method_kwargs)
+        else:
+            raise ValueError("method must be 'pts' or 'opt'.")
+
+
+def common_2beams(beams):
```

Comment:
There's a lot of clever stuff in here, and it's well-commented.  I didn't catch anything requiring changes.